Sources/MapboxCoreNavigation/PredictiveCache/PredictiveCacheManager.swift

```diff
@@ -42,19 +47,31 @@ public class PredictiveCacheManager {
     
      - parameter predictiveCacheOptions: A configuration specifying various caching parameters, such as the radii of current and destination locations.
      - parameter mapOptions: Information about `MapView` tiles such as the location and tilesets to cache. If this argument is set to `nil`, predictive caching is disabled for map tiles.
+     - parameter map: An instance of `MapboxMap`
      - seealso: `PredictiveCacheManager.init(predictiveCacheOptions:styleSourcePaths:)`
      */
-    public init(predictiveCacheOptions: PredictiveCacheOptions, mapOptions: MapOptions?) {
+    public init(predictiveCacheOptions: PredictiveCacheOptions,
+                mapOptions: MapOptions?,
+                map: MapboxMap? = nil) {
         self.predictiveCacheOptions = predictiveCacheOptions
         self.mapOptions = mapOptions
+        self.map = map
         
-        self.controllers = createControllers()
-        
+        updateControllers()
+
+        subscribeToStyleLoadEvents()
         subscribeNotifications()
     }
     
     deinit {
         unsubscribeNotifications()
+        observer?.cancel()
+    }
+
+    private func subscribeToStyleLoadEvents() {
+        self.observer = map?.onEvery(event: .styleLoaded) { [weak self] _ in
+            self?.updateControllers()
```

Comment:
Parity with [Android](https://github.com/mapbox/mapbox-navigation-android/blob/f28c864c61c576624c1682062b99798bae30f690/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt#L180)

```(kotlin)
        val onStyleLoadedListener = OnStyleLoadedListener {
        ...
        updateMapsControllers(
```
Allows to cache styles that was not loaded at the moment of PredictiveCacheManager creation